awx_collection/plugins/lookup/tower_api.py

```diff
@@ -0,0 +1,202 @@
+    required: True
+  query_params:
+    description:
+      - The query parameters to search for in the form of key/value pairs.
+    type: dict
+    required: False
+    aliases: [query, data, filter, params]
+  expect_objects:
+    description:
+      - Error if the response does not contain either a detail view or a list view.
+    type: boolean
+    default: False
+    aliases: [expect_object]
+  expect_one:
+    description:
+      - Error if the response contains more than one object.
+    type: boolean
+    default: False
+  return_objects:
+    description:
+      - If a list view is returned, promote the list of results to the top-level of list returned.
+      - Allows using this lookup plugin to loop over objects without additional work.
+    type: boolean
+    default: True
+  return_all:
+    description:
+      - If the response is paginated, return all pages.
+    type: boolean
+    default: False
+  return_ids:
+    description:
+      - If response contains objects, promote the id key to the top-level entries in the list.
+      - Allows looking up a related object and passing it as a parameter to another module.
+      - This will convert the return to a string or list of strings depending on the number of selected items.
+    type: boolean
+    aliases: [return_id]
+    default: False
+  max_objects:
+    description:
+      - if C(return_all) is true, this is the maximum of number of objects to return from the list.
+      - If a list view returns more an max_objects an exception will be raised
+    type: integer
+    default: 1000
+
+notes:
+  - If the query is not filtered properly this can cause a performance impact.
+  - In addition, the built in threshold is 10,000 items; if the query returns more an exception will be thrown.
+"""
+
+EXAMPLES = """
+- name: Load the UI settings
+  set_fact:
+    tower_settings: "{{ lookup('awx.awx.tower_api', 'settings/ui') }}"
+
+- name: Lookup any users who are admins
+  set_fact:
+    admin_user_list_view: "{{ lookup('awx.awx.tower_api', 'users', query_params={ 'is_superuser': true }, return_objects=False) }}"
+
+- name: Lookup any users who are admins and get their objects directly
+  set_fact:
+    admin_user_object: "{{ lookup('awx.awx.tower_api', 'users', query_params={ 'is_superuser': true } ) }}"
+
+- name: Lookup the admin user and fail if there are more than one, and make sure its a list (without this, the response would be an object)
+  set_fact:
+    actual_admin_user: "{{ lookup('awx.awx.tower_api', 'users', query_params={ 'username': 'admin' }, expect_one=True, wantlist=True) }}"
+
+- name: Get just the user ID of the admin user
+  set_fact:
+    admin_user_id: "{{ lookup('awx.awx.tower_api', 'users', query_params={ 'username': 'admin' }, return_ids=True, expect_one=True) }}"
+
+- name: Create a job template with a looked up credential from a folded lookup
+  tower_job_template:
+    name: "{{ job_template_name }}"
```

Comment:
these templates leave the examples as something which isn't exactly run-able either. It would be nice to be able to copy and paste it and run it

Reply:
I'm torn on this because for complex items (like a job template) we could take out the variables but if someone does not have an inventory we reference it will still fail. Unless we added "full examples" to make sure all other assets exist we run the risk of a copy/paste failing. Given that I'd almost rather see variables in there indicating that the example requires some data to function.